uint64 id
SensorSpec[] sensorSpec
CommSpec[] commSpec
AppSpec[] appSpec
